js/__internal/grids/pivot_grid/module_widget_utils.ts

```diff
@@ -59,7 +59,8 @@ function createForeachTreeFunc(isAsync) {
         foreachTreeAsyncDate = new Date();
         // @ts-expect-error
         deferred = new Deferred();
-        setTimeout(createForeachTreeAsyncHandler(deferred, i, false) as any, 0);
+        // eslint-disable-next-line @typescript-eslint/no-loop-func
+        setTimeout(() => createForeachTreeAsyncHandler(deferred, i, false) as any, 0);
```

Comment:
It is true positive from linter, isn't it? setTimeout's callback will be called after for-loop with the same value of `i`

Reply:
Yep, I agree with you 
Sorry, I hurried and did a mistake :)